codes/kotlin/chapter_stack_and_queue/linkedlist_deque.kt

```diff
@@ -64,16 +64,16 @@ class LinkedListDeque {
 
     /* 出队操作 */
     fun pop(isFront: Boolean): Int {
-        if (isEmpty()) 
+        if (isEmpty())
             throw IndexOutOfBoundsException()
         val _val: Int
         // 队首出队操作
         if (isFront) {
             _val = front!!._val // 暂存头节点值
             // 删除头节点
             val fNext = front!!.next
-            if (fNext != null) {
-                fNext.prev = null
+            fNext?.let {
```

Comment:
Is it equivalent?

Reply:
I think `fNext?.let` means `if (fNext == null) then`

Reply:
yes, it is equivalent.

Reply:
fNext ?:  means if (fNext == null) then

Reply:
Which one is correct:

1. `if (fNext == null)` euqivalent to `fNext?.let {`
2. `if (fNext != null)` euqivalent to `fNext?.let {`